src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1440,6 +1440,15 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
             } catch (AuthenticationException authenticationException) {
                 callbackHandle.onError(authenticationException);
                 return null;
+            } catch (HttpResponseException httpResponseException) {
+                // Return back the http error for silent request
+                // Try interactive flow for non-silent request.
+                if (request.isSilent()) {
+                    final AuthenticationException authenticationException = new AuthenticationException(ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, 
+                            httpResponseException.getMessage() + request.getLogInfo());
+                    callbackHandle.onError(authenticationException);
+                    return null;
```

Comment:
Because of early returns and multiple conditions it's hard to follow what happens, would be hard to maintain